admin/Makefile

```diff
@@ -1,80 +1,36 @@
-BUILD_DIR ?= .build
-VENV ?= ${BUILD_DIR}/venv
-DOCKER_TAG ?= latest
-
-GIT_HASH := $(shell git rev-parse HEAD)
-DOCKER_BASE = camptocamp/c2cgeoportal
+# Used for development purpose: please do not remove this file
 
-.PHONY: build
-build: ${BUILD_DIR}/requirements.timestamp ${BUILD_DIR}/node_modules.timestamp
-
-.PHONY: clean
-clean:
-	rm -rf ${BUILD_DIR}/*.timestamp
-
-.PHONY: cleanall
-cleanall:
-	rm -rf ${BUILD_DIR}
-	rm -rf node_modules
+DOCKER_BASE ?= camptocamp/c2cgeoportal
+DOCKER_TAG ?= latest
 
-.PHONY: test
-test: build ${BUILD_DIR}/requirements-dev.timestamp
-	# Run your project's tests.
-	${VENV}/bin/pytest --driver Chrome
+default: help
 
-.PHONY: check
-check: flake8
+.PHONY: help
+help: ## Display this help message
+	@echo "Usage: make <target>"
+	@echo
+	@echo "Possible targets:"
+	@grep -Eh '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "    %-20s%s\n", $$1, $$2}'
 
-.PHONY: flake8
-flake8: ${BUILD_DIR}/requirements-dev.timestamp
-	${VENV}/bin/flake8 c2cgeoportal_admin acceptance_tests *.py
+.PHONY: preparedev
+preparedev: ## Run this once before serve
+	make -C .. build-tools build-test-db
+	make docker-build-admin
+	docker-compose up -d db
+	docker-compose run --rm admin wait-db
+	docker-compose run --rm admin alembic --config=/opt/c2cgeoportal/commons/alembic.ini --name=main upgrade head
+	docker-compose run --rm admin alembic --config=/opt/c2cgeoportal/commons/alembic.ini --name=static upgrade head
 
 .PHONY: serve
-serve: build
-	# Run your project.
-	${VENV}/bin/pserve --reload development.ini
-
-${BUILD_DIR}/venv.timestamp:
-	# Create a Python virtual environment.
-	virtualenv -p python3 ${VENV}
-	# Upgrade packaging tools.
-	${VENV}/bin/pip install --upgrade pip==9.0.1 setuptools==36.5.0
-	touch $@
+serve: ## Run c2cgeoportal_admin in development mode
+serve:
+	docker-compose up admin
```

Comment:
I prefer using `docker-compose up -d` but I don't know ?

Reply:
The advantage here is that you directly get the logs, and you can easily restart with fresh container (Ctrl + C, up arrow, enter). For development it seems to comfortable to me.

Reply:
ok